### PR TITLE
Add v0 of HWP angle loading

### PR DESCRIPTION
**Pull request by kmharrington**
Expand the g3tsmurf loading from configs a bit and add a utility function for loading up HWP angles. This assumes the HWP angles are saved in the format generated by the `update-hwp-angles` site pipeline script. 

The loading for the HWP data is _slow_. We're going to have to fix that soon-ish. But this does work and does get the HWP angles loading into the AxisManager. We can upgrade this function when we get faster methods.

### CODE REVIEW

File: sotodlib/io/g3tsmurf_utils.py [hunk 1]
```diff
@@ -202,3 +204,41 @@ def get_batch(
...
+    )
+    time = data["hwp.hwp_angle"][0]
+    if len(time) == 0:
+        raise ValueError("No HWP data found that overlaps aman")
+    if time[0] > aman.timestamps[0] or time[-1] < aman.timestamps[-1]:
+        logger.warning(f"HWP data does not cover all of AxisManager, extrapolations" /
+                      "may be unstable")
+        
+    hwp = interp1d(data["hwp.hwp_angle"][0],data["hwp.hwp_angle"][1], bounds_error=False,
+                 fill_value="extrapolate" )
```
Comment (mhasself) [inline sotodlib/io/g3tsmurf_utils.py:242]: This sort of thing doesn't work on data with a branch cut, right?

A brute force solution is to "unwrap" the HWP angle and resample that, then put the result back in the [0, 2pi) branch.  float64 should have enough precision to handle that without adding noise, at least for ~days of data at a time.

File: sotodlib/io/g3tsmurf_utils.py [hunk 2]
```diff
@@ -202,3 +204,41 @@ def get_batch(
...
+def load_hwp_data(aman, configs=None, data_dir=None):
+    """Load HWP data that will match with aman timestamps and 
+    interpolate angles into hwp_angle
+    
+    Arguments
+    ----------
+    aman : AxisManager
+        AxisManager of loaded SMuRF data
+    configs : (optional) config filename or dictionary containing "hwp_prefix"
+    data_dir : (optional) the "hwp_prefix" data directory. overrides configs
```
Comment (msilvafe) [inline sotodlib/io/g3tsmurf_utils.py:216]: Can you point to an example `hwp_prefix` file or dictionary in the doc-string or readthedocs?

**Review by msilvafe — CHANGES_REQUESTED** [review]
The main actual comment is just addressing what Matthew brought up. I think if you address that and have tested that it works this can be merged. 

Otherwise I think it would be useful if you included in the PR comment, doc string, or readthedocs  an example of what the loading in the presence of HWP data looks like (like what's the function call) and what is the structure of the config file.

File: sotodlib/io/g3tsmurf_utils.py (hunk 1 above)
Comment (kmharrington) [reply sotodlib/io/g3tsmurf_utils.py:242]: @msilvafe  any chance you have the code for that you could just paste into this and merge into this branch? I'm more interested in connecting all the IO dots and would rather leave the math to the SAT folks.

File: sotodlib/io/load_smurf.py [hunk 3]
```diff
@@ -198,6 +198,8 @@ def from_configs(cls, configs):
         -----
         configs - dictionary containing `data_prefix` and `g3tsmurf_db` keys
         """
+        if type(configs)==str:
+            configs = yaml.safe_load( open(configs, "r"))
```
[the remark this answers is not in the document]
Comment (kmharrington) [reply sotodlib/io/load_smurf.py:202]: https://simons1.princeton.edu/confluence/display/CHILAT/SO%3A+Data+Packaging+FLP#SO:DataPackagingFLP-P10R1DataLoadingExample

File: sotodlib/io/g3tsmurf_utils.py (hunk 2 above)
Comment (kmharrington) [reply sotodlib/io/g3tsmurf_utils.py:216]: Actually, I guess you also haven't seen these changes? https://sotodlib.readthedocs.io/en/latest/hwp.html

File: sotodlib/io/g3tsmurf_utils.py [hunk 4]
```diff
@@ -30,20 +32,22 @@ def get_obs_folder(obs_id, archive):
...
-        get_obs_folder(obs_id,archive),
+        get_obs_folder(obs_id, archive),
         "outputs"
     )
 
     if not os.path.exists(path):
-        logger.error(f"Path {path} does not exist, how does {obs.obs_id} exist?")
-    return [os.path.join(path,f) for f in os.listdir(path)]
+        logger.error(f"Path {path} does not exist, " +
+                     f"how does {obs.obs_id} exist?")
```
Comment (msilvafe) [inline sotodlib/io/g3tsmurf_utils.py:49]: Also this logger is trying to return `obs.obs_id` but `obs` isn't defined.

File: sotodlib/io/g3tsmurf_utils.py [hunk 5]
```diff
@@ -53,13 +57,15 @@ def get_obs_plots(obs_id, archive):
...
-        get_obs_folder(obs_id,archive),
+        get_obs_folder(obs_id, archive),
         "plots"
     )
 
     if not os.path.exists(path):
-        logger.error(f"Path {path} does not exist, how does {obs.obs_id} exist?")
-    return [os.path.join(path,f) for f in os.listdir(path)]
+        logger.error(f"Path {path} does not exist, " +
+                     f"how does {obs.obs_id} exist?")
```
Comment (msilvafe) [inline sotodlib/io/g3tsmurf_utils.py:66]: Same here as line 49.

**Review by msilvafe — APPROVED** [review]
I'm going to go ahead and approve so that I'm not blocking merging but I would like to see this tested with the commits that I added in before it gets merged.

**Comment by kmharrington** [comment]
looks good, and I've testing it again. Thanks Max

**Pull request merged**